decorators_library/decorators.py

```diff
@@ -1 +1,66 @@
-# implement your decorators here.
+from .exceptions import TimeoutError
+import time
+import logging
+
+def timeout(integer):
```

Comment:
Problem with this one is if func takes too long, it doesn't time out until after it completes. ie, if the limit is 1 second, and func takes 10 seconds, the error is raised after 10 seconds instead of after 1 second. (also, when func _doesn't_ time out, you're not returning the return value of func.